lib/ecto/query/builder/from.ex

```diff
@@ -50,6 +50,9 @@ defmodule Ecto.Query.Builder.From do
   end
 
   defp escape_source(query, _env), do: query
+  
+  @typep dynamic_hint :: {atom, term}
+  @typep hints :: String.t() | dynamic_hint | [String.t() | dynamic_hint]
```

Comment:
I think we should consider calling `List.wrap` in `Ecto.Query.from` so we always receive a list. This way, we can do this:

```suggestion
  @typep hints :: [String.t() | dynamic_hint]
```

This simplifying both the code and the specs here.